### PR TITLE
ColorPalette: Prevent overflow of custom color button background

### PR DESCRIPTION
**Pull request by t-hamano**
## What?

This PR fixes two issues with the custom color buttons in the color palette buttons:

- The checkerboard pattern overflows at the corners
- The checkerboard pattern overlaps the button border

## Why? / How?

As for the corner overflow, it became more noticeable after the radius value was changed from 2px to 4px in #64368.

As for the overlap with the border, I think it looks more natural to shift it by 1px to avoid overlapping.

## Testing Instructions

Visit http://localhost:50240/?path=/docs/components-colorpalette--docs and check the corners of the component.

### Before

before_1
before_2

### After

after_1
after_2

### CODE REVIEW

File: packages/components/src/color-palette/style.scss [hunk 1]
```diff
@@ -41,6 +41,7 @@ $border-as-box-shadow: inset 0 0 0 $border-width rgba(0, 0, 0, 0.2);
 		background-position: 0 0, 24px 24px;
 		/*rtl:end:ignore*/
 		background-size: calc(2 * 24px) calc(2 * 24px);
+		border-radius: $radius-medium - 1 $radius-medium - 1 0 0;
```
Comment (t-hamano) [inline packages/components/src/color-palette/style.scss:44]: If we find this approach difficult to understand, we can also solve the problem by applying `overflow: hidden` to the button itself.

File: packages/components/src/color-palette/style.scss [hunk 2]
```diff
@@ -41,6 +41,7 @@ $border-as-box-shadow: inset 0 0 0 $border-width rgba(0, 0, 0, 0.2);
 		background-position: 0 0, 24px 24px;
 		/*rtl:end:ignore*/
 		background-size: calc(2 * 24px) calc(2 * 24px);
+		border-radius: $radius-medium - $border-width $radius-medium - $border-width 0 0;
```
Comment (t-hamano) [inline packages/components/src/color-palette/style.scss:41]: If we find this approach difficult to understand, we can also solve the problem by applying `overflow: hidden` to the button itself.

File: packages/components/src/color-palette/style.scss [hunk 3]
```diff
@@ -26,10 +26,10 @@ $border-as-box-shadow: inset 0 0 0 $border-width rgba(0, 0, 0, 0.2);
...
 		content: "";
 		position: absolute;
-		top: 0;
-		left: 0;
-		width: 100%;
-		height: 100%;
+		top: 1px;
+		left: 1px;
+		right: 1px;
+		bottom: 1px;
```
Comment (tyxla) [inline packages/components/src/color-palette/style.scss]: Should these use `$border-width`?

```suggestion
		top: $border-width;
		left: $border-width;
		right: $border-width;
		bottom: $border-width;
```

File: packages/components/src/color-palette/style.scss (hunk 2 above)
Comment (tyxla) [reply packages/components/src/color-palette/style.scss:41]: It actually makes a lot of sense, especially if we use `$border-width` for the `top`/`right`/`bottom`/`left`.

**Review by tyxla — APPROVED** [review]
LGTM 👍 🚀 

Just left a minor suggestion to make our intent with positioning clearer.

File: packages/components/src/color-palette/style.scss (hunk 3 above)
Comment (ciampo) [reply packages/components/src/color-palette/style.scss]: What about

```suggestion
		inset: $border-width;
```

as a shorthand?

Comment (tyxla) [reply packages/components/src/color-palette/style.scss]: Even better 👍

**Pull request merged**